Plugins/SwiftLibraryGeneratorPlugin/plugin.swift

```diff
@@ -0,0 +1,27 @@
+import PackagePlugin
+import Foundation
+
+@main
+struct SwiftLibraryGeneratorPlugin: BuildToolPlugin {
```

Comment:
Until we can support context for plugins, supporting the two variants as different plugins is the best we can do